clients/rust-legacy/tests/interest_bearing_mint.rs

```diff
@@ -281,12 +281,12 @@ fn process_instruction(
 async fn amount_conversions() {
     let rate_authority = Keypair::new();
     let mut program_test = ProgramTest::default();
-    program_test.prefer_bpf(false);
```

Comment:
So what is the reason why the tests fail if we don't prefer bpf here?

Reply:
The `prefer_bpf(false)` mode of solana-program-test uses `solana_program::program_stubs` to implement the syscalls, which no longer works when not using solana-program. So to fix the issue, we force token-2022 to run in the VM rather than the "fake native" mode of solana-program-test.